packages/headless/src/controllers/commerce/core/facets/category/headless-commerce-category-facet.ts

```diff
@@ -69,6 +82,31 @@ export function buildCategoryFacet(
   });
   const {deselectAll, isValueSelected, subscribe, toggleSelect} =
     coreController;
+  const {dispatch} = engine;
+  const getFacetId = () => coreController.state.facetId;
+  const createFacetSearch = () => {
+    return buildCategoryFacetSearch(engine, {
+      options: {facetId: getFacetId(), ...options.facetSearch},
+      select: () => {
+        dispatch(options.fetchResultsActionCreator());
+      },
+      isForFieldSuggestions: false,
+    });
+  };
+
+  const facetSearch = createFacetSearch();
+  const {state, ...restOfFacetSearch} = facetSearch;
+  const facetSearchStateSelector = createSelector(
+    (state: CommerceEngineState) => state.categoryFacetSearchSet[getFacetId()],
+    (facetSearch) => ({
+      facetSearch: {
+        isLoading: facetSearch.isLoading,
+        moreValuesAvailable: facetSearch.response.moreValuesAvailable,
+        query: facetSearch.options.query,
+        values: facetSearch.response.values,
+      },
+    })
```

Comment:
Why are we nesting the object under `facetSearch`?
```suggestion
    (facetSearch) => ({
      isLoading: facetSearch.isLoading,
      moreValuesAvailable: facetSearch.response.moreValuesAvailable,
      query: facetSearch.options.query,
      values: facetSearch.response.values,
    })
```

Reply:
No very good reason. This allows us to spread it in the state:

```
...facetSearchStateSelector(engine.state),
```

otherwise we'd do:

```
facetSearch: facetSearchStateSelector(engine.state),
```

but spreading has a (small) cost, so your suggestion is objectively better.